### PR TITLE
Refactor `SockAddrToString` with `flare::Format`

### PR DESCRIPTION
**Pull request by 4kangjc**

### CODE REVIEW

File: flare/base/net/endpoint.cc [hunk 1]
```diff
@@ -60,9 +56,7 @@ std::string SockAddrToString(const sockaddr* addr) {
       FLARE_CHECK(
           inet_ntop(af, &p->sin6_addr, result.data() + 1, result.size()));
       auto ptr = result.data() + result.find('\x00');
-      *ptr++ = ']';
-      *ptr++ = ':';
-      snprintf(ptr, kPortChars, "%d", ntohs(p->sin6_port));
+      FormatTo(ptr, "]:{}", ntohs(p->sin6_port));
```
Comment (0x804d8000) [inline flare/base/net/endpoint.cc:59]: Without a way to specify buffer size for `ptr`, this feels dangerous to me. Maybe we can instead use `inet_ntop` to format IP to a local char buffer, and then use `format` to generate the final `[ip]:port` string?

Comment (4kangjc) [reply flare/base/net/endpoint.cc]: I agree with you.

Comment (4kangjc) [reply flare/base/net/endpoint.cc]: Still need to keep `FormatTo`?

Comment (0x804d8000) [reply flare/base/net/endpoint.cc]: I’m okay with it.

**Comment by 0x804d8000** [comment]
Merged. Thanks for the patch!

**Pull request merged**